### PR TITLE
Tdl 21819 tickets stream dev changes

### PR DESCRIPTION
**Pull request by rdeshmukh15**
# Description of change
(write a short description or paste a link to JIRA)

# Manual QA steps
 - 
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tap_hubspot/__init__.py [hunk 1]
```diff
@@ -699,7 +702,74 @@ def sync_deals(STATE, ctx):
...
+            # Checking the bookmark value is present and the record is having
+            # grater than or equal to defined previous bookmark value
+            if modified_time and modified_time >= bookmark_value:
+                singer.write_record(stream_id, row, catalog.get(
+                    'stream_alias'), time_extracted=utils.now())
+                max_bk_value = modified_time
+
+    STATE = singer.write_bookmark(STATE, stream_id, bookmark_key, utils.strftime(max_bk_value))
+    singer.write_state(STATE)
+    return STATE
```
Comment (kethan1122) [inline tap_hubspot/__init__.py:769]: I don't think this return statement is required

**Review by kethan1122 — CHANGES_REQUESTED** [review]

File: tap_hubspot/__init__.py [hunk 2]
```diff
@@ -699,7 +702,74 @@ def sync_deals(STATE, ctx):
...
+    url = get_url(stream_id)
+
+    with Transformer(UNIX_MILLISECONDS_INTEGER_DATETIME_PARSING):
+        for row in gen_request_tickets(STATE, stream_id, url, params, 'results', "paging"):
+
+            modified_time_org = row[bookmark_key]
+            modified_time = utils.strptime_with_tz(datetime.datetime.strptime(
+                modified_time_org, "%Y-%m-%dT%H:%M:%S.%fZ").strftime("%Y-%m-%dT%H:%M:%SZ"))
+
+            # Checking the bookmark value is present and the record is having
```
Comment (bryantgray) [inline tap_hubspot/__init__.py:760]: Can we reword this to make it a little more clear?

Maybe something like:
> Checking the bookmark value is present on the record and it
is greater than or equal to defined previous bookmark value

File: tap_hubspot/__init__.py [hunk 3]
```diff
@@ -96,6 +96,8 @@ class StateFields:
     "forms":                "/forms/v2/forms",
     "workflows":            "/automation/v3/workflows",
     "owners":               "/owners/v2/owners",
+
+    "tickets":              "/crm/v4/objects/tickets",
```
Comment (kethan1122) [inline tap_hubspot/__init__.py:100]: Is this `v3` or `v4` ? it is mentioned as `v3` in the provided docs

> To request a list of all tickets, make a GET request to /crm/v3/objects/tickets.

Comment (rdeshmukh15) [reply tap_hubspot/__init__.py:100]: It is `v4`, if you check the API doc, the latest version of the CRM APIS, it will be `v4` 
Here, the requirement is to list the tickets with their associations and for this, I have used the endpoint `/crm/v4/objects/tickets/{ticketId}/associations/{toObjectType} ` mentioned in the endpoint section of [API doc link](https://developers.hubspot.com/docs/api/crm/tickets)

File: tap_hubspot/__init__.py (hunk 1 above)
Comment (rdeshmukh15) [reply tap_hubspot/__init__.py:769]: `return STATE` is required. If you see line no. 1111, STREAM.sync(STATE, ctx) is expecting the return value `STATE` and if I remove return statement then it will return None and that will cause the issue. Hence, keeping it as it is.

**Review by kethan1122 — APPROVED** [review]

**Review by kethan1122 — APPROVED** [review]

**Review by bryantgray — APPROVED** [review]

**Review by RushiT0122 — APPROVED** [review]
